README.md

```diff
@@ -0,0 +1,54 @@
+# fail-rs
+
+A fail points implementation for rust.
+
+## Usage
+
+First, add this to your `Cargo.toml`:
+
+```toml
+[dependencies]
+fail = "0.1"
+```
+
+Next, add this to your crate:
+
+```rust
+#[macro_use]
+extern crate fail;
+```
+
+Define fail points:
+
+```
+fn function_return_tuple() {
+    fail_point!("name1");
+}
+
+fn function_return_others() -> u64 {
+    fail_point!("name2", |r| r.map_or(2, |e| e.parse().unwrap()));
+    0
+}
+
+fn function_conditional(enable: bool) {
+    fail_point!("name3", enable, |_| {});
+}
+```
+
+Trigger a fail point via environment variable:
+
+```
+$ FAILPOINTS=bar::foo=panic cargo run
+```
+
```

Comment:
Please Add the following:
## Test

Reply:
It's not a test, it's a different usage for fail points other than the environment variables.